blog/_posts/2024-10-15-scala-3.5.0-released.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.5.0 released!
+---
+
+![Scala 3.5]({{ site.baseurl }}/resources/img/scala-3.5-launch.jpg)
+
+We are happy to announce that after the long and hard work of the entire compiler team and seven release candidates, the first version of the Scala 3.5 line is officially out!
+
+## New default runner - Scala CLI
+
+[Scala CLI](https://scala-cli.virtuslab.org/) is a popular tool among Scala devs. It allows lightning-fast running, testing, and prototyping of small Scala projects and scripts. Since 3.5.0, it has become a part of the default Scala distribution. If you install Scala through popular package managers, such as Brew or SDKMAN!, the installed `scala` command will allow you to compile, run, test, and even publish your code on Maven Central.  It will have out-of-the-box support using-directives, toolkits, compilation to native and js targets, and other goodies formerly available only if you installed a third-party tool.
+
+### Short example
+
+If we have the following file named `biggerThan.scala`
+
+```scala
+//> using dep com.lihaoyi::os-lib:0.10.3
+
+@main def run(path: String, size: Int) =
+  os.list(os.Path(path, os.pwd))
+    .filter: p =>
+      os.size(p) > size * 1024
+    .foreach(println)
+```
+
+We can run `scala biggerThan -- ../my-directory 10`. This will download the os-lib and all its transitive dependencies, then compile the source file, and finally run the compiled program, printing all files in `../my-directory` bigger than 10 kB. Subsequent invocations of the command will use cached binary, or in case of changes in the file, trigger incremental compilation.
```

Comment:
```suggestion
We can run `scala biggerThan -- ../my-directory 10`. This will download os-lib and its transitive dependencies, compile the source code, and finally run the compiled program, printing all files in `../my-directory` bigger than 10 kB. Subsequent invocations of the command will use cached bytecode, or if the source code changed, trigger incremental compilation.
```